### ISSUE TITLE
If the "Enable the Media Library" option is unchecked in "Navigation Settings", should UMS be scanning all media on startup?

### ISSUE THREAD

**Issue by esabol** [UniversalMediaServer/UniversalMediaServer#1960]
I'm an old school PMS/UMS user and don't use the newfangled Media Library feature that all the young kids rave about. I have it turned off in the "Navigation Settings" pane. In spite of this, UMS still scans every single media file in my "Shared Content" directories whenever I restart UMS, and it takes *hours* to get through all of my media.

I'm under the impression that this scanning is primarily for the Media Library feature. Is that correct?

If the Media Library feature is turned off, would it be possible to avoid scanning all of my media whenever I restart UMS? I'm pretty sure that's how older versions of UMS worked and that this changed sometime in the past 12 months.

**Comment by esabol** [comment]
Actually, I think this might have been fixed in 9.4.0? Closing.

**Comment by SubJunk** [comment]
@esabol I don't think it changed in 9.4.0, but the cache feature works outside of the Media Library too - it makes browsing all folders faster, and it has been working like that since PMS days :)
The thing that changed in the 7.0.0 release is just that we scan the folders on startup.
The scan can be disabled on the `Shared Content` tab, without disabling the entire cache feature. That will mean things are cached as you access them instead of in advance, so you will still get the benefits on your second access

**Comment by esabol** [comment]
> @esabol I don't think it changed in 9.4.0...

Well, something seems to have changed. What does “Fixed broken cache loading in some situations” from the 9.4.0 release notes mean? Maybe that’s the difference. In any case, it’s not doing it anymore. Thanks!

**Comment by SubJunk** [comment]
@esabol that change refers to a fix for if users started an instance of UMS while something else was using the database/cache (often a previous instance of UMS), we would fail to load the cache and then disable the cache permanently for the user, unless they changed the setting back. Now in that situation, we wait for 10 seconds if the load fails, and try again, and if it fails that time we display an alert about it.

**Comment by esabol** [comment]
Ugh. It's doing it again. Version 9.8.2. "Scan shared folders at startup" is NOT checked and all Media Library options are disabled, but it's still clearly scanning all of my drives again.

**Comment by SubJunk** [comment]
Can you please attach logs for this? I can't reproduce

**Comment by esabol** [comment]
My debug.log file is 1.76 GB. I doubt you want it all.

**Comment by valib** [comment]
@esabol please delete the database start the UMS and after e.g. 5 or 10 minutes stop the UMS and send the log.

**Comment by esabol** [comment]
I can't seem to reproduce it after restarting UMS, of course. I think it's actually initiated by my TV? It's hard to tell, but I think this is where it seems to start scanning the whole drive:
```
TRACE 2020-11-25 00:48:36.470 [HTTPv2 Request Worker 12] Stripping preceding slash from: /upnp/control/content_directory
TRACE 2020-11-25 00:48:36.470 [HTTPv2 Request Worker 12] Matched media renderer "[LG] webOS TV OLED65C9PUA" based on address 192.168.1.21
DEBUG 2020-11-25 00:48:36.470 [HTTPv2 Request Worker 12] Recognized media renderer "[LG] webOS TV OLED65C9PUA"
TRACE 2020-11-25 00:48:36.472 [HTTPv2 Request Worker 12] Received a browse request from [LG] webOS TV OLED65C9PUA [LG OLED] (192.168.1.21:36178):

POST /upnp/control/content_directory HTTP/1.1

HEADER:
  SOAPAction: "urn:schemas-upnp-org:service:ContentDirectory:1#Browse"
  DLNADeviceName.lge.com: %5bLG%5d%20webOS%20TV%20OLED65C9PUA
  User-Agent: Linux/4.4.84-169.gld4tv.4 UPnP/1.0 LGE WebOS TV LGE_DLNA_SDK/1.6.0/05.00.03 DLNADOC/1.50
  Host: 192.168.1.101:5002
  Content-Length: 676
  Content-Type: text/xml; charset="utf-8"

CONTENT:
<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
  <s:Body>
    <u:Browse xmlns:u="urn:schemas-upnp-org:service:ContentDirectory:1">
      <ObjectID>68</ObjectID>
      <BrowseFlag>BrowseDirectChildren</BrowseFlag>
      <Filter>res@resolution,res@nrAudioChannels,res@sampleFrequency,res@bitrate,dc:creator,res@dlna:cleartextSize,dc:date,upnp:genre,res,res@duration,res@size,upnp:albumArtURI,upnp:originalTrackNumber,upnp:album,upnp:artist,upnp:author</Filter>
      <StartingIndex>0</StartingIndex>
      <RequestedCount>30</RequestedCount>
      <SortCriteria/>
    </u:Browse>
  </s:Body>
</s:Envelope>

TRACE 2020-11-25 00:48:37.478 [HTTPv2 Request Worker 12] Could not match any format to "/Volumes/Coruscant/Movies/.DS_Store"
TRACE 2020-11-25 00:48:37.479 [HTTPv2 Request Worker 12] Matched format MKV to "/Volumes/Coruscant/Movies/foo.mkv"
TRACE 2020-11-25 00:48:37.480 [HTTPv2 Request Worker 12] Matched format MKV to "/Volumes/Coruscant/Movies/bar.mkv"
etc.
```
It then proceeds to scan the whole drive and then my other drives. After some time, I start to see thousands of lines like these:
```
DEBUG 2020-11-25 01:19:57.850 [GlobalId cleanup] deleting invalid id 30857
DEBUG 2020-11-25 01:19:57.850 [GlobalId cleanup] deleting invalid id 30856
DEBUG 2020-11-25 01:19:57.850 [GlobalId cleanup] deleting invalid id 30855
DEBUG 2020-11-25 01:19:57.850 [GlobalId cleanup] deleting invalid id 30854
DEBUG 2020-11-25 01:19:57.850 [GlobalId cleanup] deleting invalid id 30853
DEBUG 2020-11-25 01:19:57.850 [GlobalId cleanup] deleting invalid id 30852
```

**Comment by SubJunk** [comment]
Ah yes, this is a known problem with LG TVs, they like to browse all of the server folders by themselves. Unfortunately we have no way to know whether they are legitimately browsing (because a person is doing it), or if the TV is doing it automatically.
I don't know why they do it. Maybe there is a setting for it on the TV? Or maybe contact LG support about it?
